src/adapters/book-adapters.ts

```diff
@@ -0,0 +1,245 @@
+ * @param {String} bookInput.status - Book's status.
+ * @param {Buffer} bookInput.image - Book's image(optional).
+ * @param {String} bookInput.description - Book's description.
+ * @param {String} bookInput.genre - Book's genre.
+ * @param {String} bookInput.author - Book's author.
+ * @example
+ * const bookInput = {
+ *  title = "Pachinko",
+ *  language = "English",
+ *  ageRange = "Adults",
+ *  publishingYear = 2017,
+ *  status = "Open",
+ *  description = "Pachinko is the second novel by Harlem-based author and journalist Min Jin Lee.
+ *  Published in 2017, Pachinko is an epic historical fiction novel following a Korean family who immigrates to Japan.",
+ *  genre = "Historical Fiction",
+ *  author = "Min Jin Lee"
+ * };
+ * export const createBookAdapter = async (bookInput) => {
+ *  // function implementation here
+ * };
+ * @returns {Promise<Object>} A promise that resolves book's creation information.
+ */
+export const createBookAdapter = async(bookInput:bookInput) =>{
+    const url = 'http://localhost:8000/api/v1/books'
+    const data = await fetchAPIData(url, 'POST', bookInput )
+    return data
+    
+}
+
+// Get all books for all users and all books owner adapter
+type queryBook = { 
+    title?:string,
+    author?:string, 
+    sort?:string, 
+    fields?:string, 
+    searchRadius?:number, 
+    latitude?: number, 
+    longitude?: number
+}
+
+type booksSchema ={
+     _id: string,
+    title: string,
+    language: string,
+    ageRange: string,
+    publishingYear: number,
+        owner: {
+            _id: string,
+            username: string,
+            latitude: number,
+            longitude: number,
+        },
+    status: string,
+    description: string,
+    genre: string,
+    author: string,
+    createdAt: string,
+    updatedAt: string,
+    __v: number,
+    imageURL?: string
+}
+/**
+ * This function will get all books for users with the provided information
+ * @param {String} queryBook - The query object containing information for query parameters.
+ * @param {String} queryBook.title - Book's title as query parameter.
+ * @param {String} queryBook.author - Book's author as query parameter.
+ * @param {String} queryBook.sort - Sort as query parameter.
+ * @param {String} queryBook.fields - Fields as query parameter.
+ * @param {Number} queryBook.searchRadius - Search radius as query parameter.
+ * @param {Number} queryBook.latitude - Latitude as query parameter.
+ * @param {Number} queryBook.longitude - Longitude as query parameter.
+ * @example
+ * const queryBook = {
+ *  title = "Pachinko",
+ *  author = "Min Jin Lee",
+ *  sort = "CreatedAt",
+ *  fields = "Genre",
+ *  searchRadius = 25,
+ *  latitude = 20,
+ *  longitude = 40
+ * };
+ * export const getAllBooksAdapter = async(queryBook) => {
+ * // function implementation here
+ * };
+ * @returns {Promise<Object>} A promise that resolves get all books(for users) information.
```

Comment:
So this function will return an array not object, right? 
@returns {Promise<booksSchema[ ]>}

Reply:
Thanks for pointing this out Manizha,  will fix it soon.